File: src/api/stubs/packages.js
'use strict';

const express = require('express');
const router = express.Router();

/**
 * GET /api/packages - 요금제 패키지 목록
 * constants.js와 동일한 패키지 정의를 반환 (단일 진실 소스)
 */
router.get('/', (req, res) => {
  const packages = {
    'lite': {
      name: 'Lite',
      displayName: '라이트',
      basePrice: 3.30,
      priceUSD: 3.63,
      baseUsage: 330000,
      bonusRate: 0.09,
      bonusUsage: 29700,
      totalUsage: 359700,
      expireDays: 30,
      autopayBonusRate: 0.09,
      target: '개인/취미'
    },
    'standard': {
      name: 'Standard',
      displayName: '스탠다드',
      basePrice: 11.00,
      priceUSD: 12.10,
      baseUsage: 1100000,
      bonusRate: 0.07,
      bonusUsage: 77000,
      totalUsage: 1177000,
      expireDays: 30,
      autopayBonusRate: 0.09,
      target: '일반 사용자'
    },
    'pro': {
      name: 'Pro',
      displayName: '프로',
      basePrice: 22.00,
      priceUSD: 24.20,
      baseUsage: 2200000,
      bonusRate: 0.05,
      bonusUsage: 110000,
      totalUsage: 2310000,
      expireDays: 30,
      autopayBonusRate: 0.09,
      target: '헤비 유저'
    },
    'max': {
      name: 'Max',
      displayName: '멕스',
      basePrice: 44.00,
      priceUSD: 48.40,
      baseUsage: 4400000,
      bonusRate: 0.03,
      bonusUsage: 132000,
      totalUsage: 4532000,
      expireDays: 30,
      autopayBonusRate: 0.09,
      target: '헤비 유저/기업'
    }
  };

  res.json({
    success: true,
    packages,
    count: Object.keys(packages).length,
    timestamp: new Date().toISOString()
  });
});

module.exports = router;

File: src/app/api/v1/routers/packages.py
"""
구독 패키지 API 라우터
"""
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from src.app.infrastructure.db.session import get_db
from src.app.infrastructure.db.models import SubscriptionPackage

router = APIRouter(prefix='/api/packages', tags=['packages'])


def _package_to_dict(pkg) -> dict:
  """패키지 모델을 dict로 변환"""
  return {
    'package_type': pkg.package_type,
    'name': pkg.name,
    'display_name': pkg.display_name,
    'base_price': float(pkg.base_price) if pkg.base_price else 0,
    'price_usd': float(pkg.price_usd) if pkg.price_usd else 0,
    'base_usage': pkg.base_usage,
    'bonus_rate': float(pkg.bonus_rate) if pkg.bonus_rate else 0,
    'bonus_usage': pkg.bonus_usage,
    'total_usage': pkg.total_usage,
    'expire_days': pkg.expire_days,
    'target': pkg.target,
    'sort_order': pkg.sort_order,
    'is_popular': pkg.is_popular,
  }


@router.get('/')
async def get_packages(db: AsyncSession = Depends(get_db)):
  """모든 활성 패키지 조회"""
  result = await db.execute(
    select(SubscriptionPackage)
    .where(SubscriptionPackage.is_active == True)
    .order_by(SubscriptionPackage.sort_order)
  )
  packages = result.scalars().all()
  return {
    'success': True,
    'data': {'packages': [_package_to_dict(p) for p in packages]},
    'error': None,
  }


@router.get('/{package_type}')
async def get_package(package_type: str, db: AsyncSession = Depends(get_db)):
  """특정 패키지 상세"""
  result = await db.execute(
    select(SubscriptionPackage)
    .where(SubscriptionPackage.package_type == package_type)
  )
  pkg = result.scalar_one_or_none()
  if not pkg:
    raise HTTPException(status_code=404, detail=f'패키지 없음: {package_type}')
  return {
    'success': True,
    'data': {'package': _package_to_dict(pkg)},
    'error': None,
  }
